src/main/cli/cli.c

```diff
@@ -1362,6 +1362,66 @@ static void cliSerialPassthrough(char *cmdline)
 }
 #endif
 
+#if defined(USE_ESC_SENSOR)
+static void cliEscSensorPassthrough(char *cmdline)
+{
+    uint32_t useEscSensorBaud = 115200;
+    if (!isEmpty(cmdline)) {
+        useEscSensorBaud = atoi(cmdline);
+    }
+    
+    const serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_ESC_SENSOR);
+    if (!portConfig) {
+        cliPrintLine("no esc sensor port configured.");
+        return;
+    }
+    
+    serialPortUsage_t *serialPortUsage = findSerialPortUsageByIdentifier(portConfig->identifier);
+    serialPortUsage->function = FUNCTION_NONE;
+    serialPortUsage->serialPort = NULL;
+    
+    serialPort_t *passThroughPort;
+    passThroughPort = openSerialPort(portConfig->identifier,
+        FUNCTION_NONE,
+        NULL,
+        NULL,
+        useEscSensorBaud,
+        MODE_RXTX,
+        (SERIAL_STOPBITS_1 | SERIAL_PARITY_NO | SERIAL_BIDIR | SERIAL_BIDIR_PP)
+    );
+    
+    if (!passThroughPort) {
+        cliPrintLine("esc sensor port could not be opened.");
+        return;
+    }
+    
+    pwmDisableMotors();
+    pwmOutputPort_t *pwmMotors = pwmGetMotors();
+    for (volatile uint8_t i = 0; i < MAX_SUPPORTED_MOTORS; i++) {
```

Comment:
`getMotorCount()` should be used here instead of `MAX_SUPPORTED_MOTORS`.

Reply:
will try

Reply:
It should - then you can simply drop DTR in your custom ESC flashing app when you are done flashing, and the flight controller will reboot.

On a side note: We should probably look into making this available as an MSP command as well - then you will be able to integrate the entire process, from setting up the serial passthrough, flashing, to rebooting into the firmware again after flashing is done.

Reply:
setting up the pasthrough already works with using cli :D see:
https://cdn.discordapp.com/attachments/520684572275507222/589430165751463937/VID-20190615-WA0024.mp4

Reply:
Yeah, but that is super hacky, and can easily and without warning be broken by changes in CLI, which is why this is discouraged.

Reply:
ok. if you say a MSP command would be better, we can close this pull request and i will start to do a MSP command :)

Reply:
I thought of doing both, like we do for the `escprog` commands that can be invoked from CLI or directly through MSP - some manufacturers (not KISS AFAIK) use this to allow programming / flashing of ESCs without requiring the user to fiddle with CLI.
What we can do is that we add this to CLI for now, and once it's done then I'll look into adding it as an MSP command as well. @azolyoung might be interested in this as well.

Reply:
thats fine for me :)
for now i will have our GUI handling the CLI. and later we can switch to MSP commands.
we are close to release and im happy with all things i can get running till then ;)